src/Microsoft.DocAsCode.Build.Engine/PostProcessors/SitemapGenerator.cs

```diff
@@ -0,0 +1,167 @@
+        private const string HtmlExtension = ".html";
+        private const string SitemapName = "sitemap.xml";
+
+        public string Name => nameof(SitemapGenerator);
+
+        public ImmutableDictionary<string, object> PrepareMetadata(ImmutableDictionary<string, object> metadata)
+        {
+            return metadata;
+        }
+
+        public Manifest Process(Manifest manifest, string outputFolder)
+        {
+            if (string.IsNullOrEmpty(manifest.SitemapOptions?.BaseUrl))
+            {
+                return manifest;
+            }
+
+            if (!manifest.SitemapOptions.BaseUrl.EndsWith("/"))
+            {
+                manifest.SitemapOptions.BaseUrl += '/';
+            }
+
+            if (!Uri.TryCreate(manifest.SitemapOptions.BaseUrl, UriKind.Absolute, out var baseUri))
+            {
+                Logger.LogWarning($"Base url {manifest.SitemapOptions.BaseUrl} is not in a valid uri format.");
+                return manifest;
+            }
+
+            if (manifest.SitemapOptions.Priority.HasValue && (manifest.SitemapOptions.Priority < 0 || manifest.SitemapOptions.Priority > 1))
+            {
+                Logger.LogWarning($"Invalid priority {manifest.SitemapOptions.Priority}, priority must be between 0.0 and 1.0. Use default value 0.5 instead");
+                manifest.SitemapOptions.Priority = 0.5;
+            }
+
+            XElement urlset = new XElement(XName.Get("urlset", NamespaceName));
+            XDocument sitemapDocument = new XDocument(urlset);
+
+            foreach (var file in GetHtmlOutputFiles(manifest).OrderBy(s => s.Item1))
+            {
+                var options = GetOptions(manifest.SitemapOptions, file.Item1);
+
+                var currentBaseUri = baseUri;
+                if (options.BaseUrl != manifest.SitemapOptions.BaseUrl)
+                {
+                    if (!Uri.TryCreate(options.BaseUrl, UriKind.Absolute, out currentBaseUri))
+                    {
+                        Logger.LogWarning($"Base url {options.BaseUrl} is not in a valid uri format, use base url from the default setting {manifest.SitemapOptions.BaseUrl} instead.");
+                        currentBaseUri = baseUri;
+                    }
+                }
+
+                urlset.Add(GetElement(file.Item2.RelativePath, currentBaseUri, options));
+            }
+
+            var sitemapOutputFile = Path.Combine(outputFolder, SitemapName);
+            Logger.LogInfo($"Sitemap file is successfully exported to {sitemapOutputFile}");
+            sitemapDocument.Save(sitemapOutputFile);
+            return manifest;
+        }
+
+        private IEnumerable<Tuple<string, OutputFileInfo>> GetHtmlOutputFiles(Manifest manifest)
+        {
+            if (manifest.Files == null)
+            {
+                yield break;
+            }
+
+            foreach(var file in manifest.Files)
+            {
+                if (file.DocumentType != "Toc"
```

Comment:
>file.DocumentType != "Toc" [](start = 20, length = 26)

also exclude non html resources?

Reply:
The next `file.OutputFiles.TryGetValue(HtmlExtension` do it